lib/table.js

```diff
@@ -111,22 +111,33 @@ var Table = Class.extend({
             done(null, result);
         });
     },
-    _updateRecord: function(recordId, recordData, opts, done) {
-        var record = new Record(this, recordId);
-        if (!done) {
-            done = opts;
-            record.patchUpdate(recordData, done);
-        } else {
-            record.patchUpdate(recordData, opts, done);
-        }
-    },
-    _replaceRecord: function(recordId, recordData, opts, done) {
-        var record = new Record(this, recordId);
-        if (!done) {
-            done = opts;
-            record.putUpdate(recordData, done);
+    _updateRecords: function(isDestructiveUpdate, recordsOrRecordId, recordDataOrOptsOrDone, optsOrDone, done) {
+        var opts;
+
+        if (isArray(recordsOrRecordId)) {
+            var that = this;
+            var records = recordsOrRecordId;
+            opts = isPlainObject(recordDataOrOptsOrDone) ? recordDataOrOptsOrDone : {};
+            done = optsOrDone || recordDataOrOptsOrDone;
+
+            var method = isDestructiveUpdate ? 'put' : 'patch';
+            var requestData = assign({records: records}, opts);
+            this._base.runAction(method, '/' + this._urlEncodedNameOrId() + '/', {}, requestData, function(err, resp, body) {
+                if (err) { done(err); return; }
+
+                var result = body.records.map(function (record) {
+                    return new Record(that, record.id, record);
+                });
+                done(null, result);
+            });
         } else {
-            record.putUpdate(recordData, opts, done);
+            var recordId = recordsOrRecordId;
+            var recordData = recordDataOrOptsOrDone;
+            opts = isPlainObject(optsOrDone) ? optsOrDone : {};
+            done = done || optsOrDone;
+
+            var recordMethod = isDestructiveUpdate ? 'putUpdate' : 'patchUpdate';
+            new Record(this, recordId)[recordMethod](recordData, opts, done);
```

Comment:
This feels a bit magical and would break if people are using aggressive minification (I don't know if aggressive minification works with other parts of Airtable.js at the moment, but let's not get further away). What do you think about:

```
var record = new Record(this, recordId);
if (isDestructiveUpdate) {
    record.putUpdate(recordData, opts, done);
} else {
    record.patchUpdate(recordData, opts, done);
}
```